CMakeLists.txt

```diff
@@ -80,68 +76,31 @@ if(BUILD_TESTING)
     enable_testing()
 endif()
 
-set(ALL_TARGETS fex visualboyadvance-m vbamcore vbam)
-
-add_custom_target(generate)
+include(GNUInstallDirs)
+include(Options)
+include(Architecture)
+include(Toolchain)
 
 #Output all binaries at top level
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR})
 
 if(NOT HTTPS)
-    add_definitions(-DNO_HTTPS)
-endif()
-
-if(WIN32 OR CMAKE_TOOLCHAIN_FILE MATCHES "[Mm][Ii][Nn][Gg][Ww]")
-    # Win32 deps submodule
-    set(git_checkout FALSE)
-
-    if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
-        set(git_checkout TRUE)
-        execute_process(
-            COMMAND git submodule update --init --remote --recursive
```

Comment:
I still want it to do this automatically somewhere on Windows.

Reply:
I moved it to be done before the project is configured for windows and mingw.